source/BufferParser.cpp

```diff
@@ -89,7 +90,7 @@ void BufferParser::addProtectionMetadataToSegment(std::unique_ptr<IMediaPipeline
         size_t subSampleCount = metadata.subsamples.size();
         for (size_t subSampleIdx = 0; subSampleIdx < subSampleCount; ++subSampleIdx)
         {
-            GST_DEBUG("SUBSAMPLE: %d/%d C: %d E: %d", subSampleIdx, subSampleCount,
+            GST_DEBUG("SUBSAMPLE: %lu/%lu C: %d E: %d", (long unsigned int) subSampleIdx, (long unsigned int) subSampleCount,
```

Comment:
Please consider using `%zu` which should work without casting to `long unsigned int`.